src/util/log/fd_log.c

```diff
@@ -464,17 +464,109 @@ void fd_log_level_stderr_set ( int level ) { FD_VOLATILE( fd_log_private_level_s
 void fd_log_level_flush_set  ( int level ) { FD_VOLATILE( fd_log_private_level_flush   ) = level; }
 void fd_log_level_core_set   ( int level ) { FD_VOLATILE( fd_log_private_level_core    ) = level; }
 
+/* Total size of the log buffer (fd_log_private_log_msg) */
+
+#define FD_LOG_BUF_SZ (4UL*4096UL)
+
+/* Log buffer, used by fd_log_private_0 and fd_log_private_hexdump_msg */
+
+static FD_TLS char fd_log_private_log_msg[ FD_LOG_BUF_SZ ];
+
+/* Add into the log buffer (with *printf() formatting) at the current offset and then increment 
+   the pointer accordingly */
+
+#define FD_LOG_HEXDUMP_ADD_TO_LOG_BUF(...)  do {                              \
+    num_bytes_written = sprintf ( log_buf_ptr, __VA_ARGS__ );                 \
```

Comment:
Doesn't bother me too much in that the usage should already bound this.  I would put this macro inside the hexdump function itself though (and undef it at the end) as such is the only place it is used and it implicitly assumes the log_buf_ptr variable that is declared there (with the same scope).  That would then also prevent any use cases where the sprintf use wasn't already bounded.

Reply:
@ripatel-jump I totally understand your preference for `snprintf` - I'm security paranoid myself, being a security guy as my day job, principally in C and C++ code review and blackbox security review (reverse engineering). The reason I chose to go with the `sprintf` API in this context, however, was because a) we can be sure due to the boundary checks (and overall usage context) in `fd_log_private_hexdump_msg` that we'll never extend beyond the bounds of `fd_log_private_log_msg` and corrupt memory, and b) `snprintf` returns the number of characters that *would've* be written if the size were unlimited, rather than the number of characters actually written as `sprintf` does.

If it's satisfactory to you guys, then, I'll simply limit the scope of usage for `FD_LOG_HEXDUMP_ADD_TO_LOG_BUF` by `#define`'ing and `#undef`'ing within `fd_log_private_hexdump_msg`, to be sure that no one ever uses this macro outside of the intended use case.